handlers/main.yml

```diff
@@ -1,10 +1,11 @@
 ---
-- name: enable and restart vpn services
-  service:
-    name: "{{ item }}"
-    state: restarted
-    enabled: yes
-  loop: "{{ __vpn_services }}"
-  async: 10
-  poll: 0
+- name: restart vpn service and wait for ssh conn to return
+  include_tasks: tasks/enable_restart_vpn.yml
   listen: __vpn_handler_enable_start_vpn
+
+- name: send pings to initialize mesh connections
+  command: "ping -c 5 {{ item }}"
+  loop: "{{ ansible_play_batch }}"
+  when: item != inventory_hostname
+  ignore_errors: true
```

Comment:
is there some way you can distinguish a ping that fails because the service hasn't been restarted yet vs. a ping that fails due to some other reason?  If so, then you could catch the error with a `register: __vpn_register_ping`, then check for the "hard" failure condition with a `failed_when: ... some condition of __vpn_register_ping`
In general it isn't good to use `ignore_errors: true` as it will mask failures we want to be concerned about - however, this may be one of those cases like https://github.com/linux-system-roles/metrics/blob/master/roles/elasticsearch/tasks/main.yml#L62 that we have to use `ignore_errors: true`

Reply:
I think that the `ignore_errors: true` is unnecessary now (since we are waiting for the SSH connection to return anyway in an earlier task). I will remove it. Some ping packets may be lost due to the fact that ipsec is not fully initialized yet, but that doesn't cause an ansible failure. Users will be able to see the output of the ping commands in the controller console.